shared/src/main/scala/weaponregex/WeaponRegeX.scala

```diff
@@ -11,6 +11,7 @@ import scala.scalajs.js.annotation._
 @JSExportTopLevel("WeaponRegeX")
 object WeaponRegeX {
 
+  // Can't these also be default arguments instead of 4 separate methods?
```

Comment:
We splitted the API to be platform-specfic now. For the Scala version, these will be combined, but I think you cannot provide named parameters in a function call in JS.